src/Examine.AzureDirectory/AzureDirectory.cs

```diff
@@ -287,17 +287,24 @@ public override long FileLength(string name)
                 return CacheDirectory.FileLength(name);
             }
 
-            var blob = _blobContainer.GetBlockBlobReference(RootFolder + name);
-            blob.FetchAttributes();
+            try
+            {
```

Comment:
Should this try/catch just be around the GetBlockBlobReference/FetchAttributes calls, it doesn't seem to be needed around all of that logic?

Reply:
I'm looking into a few other issues regarding this and getting some tests in place. I'll go ahead and merge this and can fix up some things after. Thanks!